ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/AddMetadata.java

```diff
@@ -108,6 +78,40 @@
         });
   }
 
+  /** Merge a JSON byte payload with a ObjectNode. */
+  public static byte[] merge(byte[] payload, ObjectNode metadataObject)
+      throws UncheckedIOException {
```

Comment:
This signature feels unbalanced. I think I'd prefer that both arguments be byte arrays and it's expected that the caller has to transform the ObjectNode into bytes. I don't have a strong rationale for this, though.

And I think I'd call this `mergedPayload` to indicate what it returns.

Reply:
A consistent interface is rationale enough. I went from `mergePayloadWithMetadata` to `merge`, but it seems like I missed out on the happy medium.

Reply:
I renamed the function `mergePayload`, which seems to fit the naming convention a bit better.